### DISCUSSION TITLE
Help with reimporting music and extra files

### DISCUSSION THREAD

**Discussion by sindreruud** [beetbox/beets#4925] [Q&A]
Hi!

I started using Beets a couple of years ago. I am quite happy with my setup, but always looking to improve it. 

In the beginning I didn't take much care to match my music to the exact release I actually have (digital, cd, vinyl, label, catalog-number, reissues, etc). As long as the artist, album, and track name matched, I was happy. Now my OCD is kicking in and it has started to annoy me slightly, and I am looking to improve this.

Most of my music is in `/mnt/downloads/album-name`, and then I import it (copy) to `/mnt/music/artist/album-name`. So I still have all those original and untouched files in `/mnt/downloads`. 

Recently I discovered [gazelle-origin](https://github.com/x1ppy/gazelle-origin). A really amazing tool that creates a .yaml for your albums, stored inside the albumfolder. The file contains for instance these lines:
```
Media:                   CD
Edition year:            2014
Record label:            Atlantic
Catalog number:          8122795828  
Edition:                 Remastered
```
Now you can install the other great tool, [beets-originquery](https://github.com/x1ppy/beets-originquery) and beets will use this yaml file to help you match with the exact release you have. It also shows the info in the cli so you have quick access to it if you have to match something manually: 
```
$> beet import ~/music
/home/x1ppy/music/(1973) Houses Of The Holy [2014 Remaster] (8 items)
originquery: Using origin file /home/x1ppy/music/(1973) Houses Of The Holy [2014 Remaster]/origin.txt
originquery: ╔════════════════╤═════════════╤═════════════╗
originquery: ║ Field          │ Tagged Data │ Origin Data ║
originquery: ╟────────────────┼─────────────┼─────────────╢
originquery: ║ Media          │             │ CD          ║
originquery: ║ Edition year   │ 1973        │ 2014        ║
originquery: ║ Record label   │             │ Atlantic    ║
originquery: ║ Catalog number │             │ 8122795828  ║
originquery: ║ Edition        │             │ Remastered  ║
originquery: ╚════════════════╧═════════════╧═════════════╝
Correcting tags from:
    Led Zeppelin - Houses Of The Holy
To:
    Led Zeppelin - Houses of the Holy
URL:
    https://musicbrainz.org/release/51a4e8b4-1af1-4daf-a746-ac1c7206dd02
(Similarity: 100.0%) (CD, 2014, XE, Atlantic, 8122795828)
 * The Song Remains The Same   -> The Song Remains the Same
 * Over The Hills And Far Away -> Over the Hills and Far Away
 * D'yer Mak'er                -> D’yer Mak’er
[A]pply, More candidates, Skip, Use as-is, as Tracks, Group albums, Enter search, enter Id, aBort?
```

Aweseom! Highly recommended!

However, my problem is that I only managed to populate all of my albumfolders in `/mnt/downloads` with yaml files. As my actual beets library in `/mnt/music` is renamed and reorganized when imported, there is no link, and I see no easy way to bring these yaml files into my music-folder.

So I think maybe my best course of action would be to import from `/mnt/downloads` again, beets would probably give me the duplicate warning:
```
This album is already in the library!
[S]kip new, Keep all, Remove old, Merge all?
```
At which point I guess i select `Remove old`. But I don't actually need to copy the files over, I just need it to fix the metadata, and write it to the files. 

Also I will probably add [beets-copyartifacts](https://github.com/adammillerio/beets-copyartifacts) to my config, so it copies over the `origin.yaml` file to my beets library.

Any thoughts on how to accomplish this, except the import remove old strategy?

**Comment by sampsyo** [comment]
That does sound like an awesome solution in general!

Good question about how best to do this to your existing library. To be perfectly honest, I'd be tempted to `Remove old` and just replace the old files with new ones… unless that poses a particular risk you're worried about (which is plausible). Another option could be to cook up some out-of-band solution for putting the origin YAML files into your current directory structure, and then re-import your library in place. (Re-imports just update existing metadata for stuff already in your library.) Doing this perfectly could be hard, but maybe a script with some simple heuristics could get most of them in the right place?

**Reply by sindreruud** [reply]
Thanks for chiming in! 

Some script could be a good solution. Probably not too hard to use some regex to match the folder name in my downloads folder to the album folder in my beets library. 

Come to think of it, the .yaml file includes the album-name, so that would make the script even easier. Copy the yaml file over, then do an inplace reimport to update Metadata. 

If I know myself I will probably spend as long making this script as I would have used just importing/`remove old` though. But where is the fun in that! 

Thanks again :)
